pkg/ui/Flow.tsx

```diff
@@ -117,29 +116,56 @@ export class Flow<T extends Values> extends Component<Props<T>, State<T>> {
   }
 
   // Handles form submission
-  handleSubmit = (e: MouseEvent | FormEvent) => {
+  handleSubmit = (event: FormEvent<HTMLFormElement> | MouseEvent) => {
     // Prevent all native handlers
-    e.stopPropagation()
-    e.preventDefault()
+    event.stopPropagation()
+    event.preventDefault()
 
     // Prevent double submission!
     if (this.state.isLoading) {
       return Promise.resolve()
     }
 
+    const form = event.currentTarget
+
+    let body: T | undefined
+
+    if (form && form instanceof HTMLFormElement) {
+      const formData = new FormData(form)
+
+      // map the entire form data to JSON for the request body
+      body = Object.fromEntries(formData) as T
+
+      if ("nativeEvent" in event) {
```

Comment:
`nativeEvent` is always part of this `event` type

Reply:
true, will remove it